driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -1451,6 +1457,11 @@ protected void initChannel(SocketChannel channel) throws Exception {
 
             // pipeline.addLast("debug", new LoggingHandler(LogLevel.INFO));
 
+            if (metrics != null) {
+                pipeline.addLast("inboundTrafficMeter", new InboundTrafficMeter(metrics.getBytesReceived()));
+                pipeline.addLast("outboundTrafficMeter", new OutboundTrafficMeter(metrics.getBytesSent()));
+            }
+
```

Comment:
Open question: should we include SSL traffic in those metrics?

On the one hand, we're interested about the impact of our data model and queries on the traffic, so SSL is a bit off topic. On the other, I can see how some people would expect this to be the overall traffic.

Reply:
According to [this SO question](https://stackoverflow.com/questions/548029/how-much-overhead-does-ssl-impose) the overhead in bytes sent/received of SSL vs non SSL is basically confined to the sole handshake phase (of course there is a lot of CPU overhead, but that's irrelevant here). Since these meters are never reset, I'd say that this overhead would tend to zero over time, so it doesn't really matter much.

Reply:
ok